astarte/device/exceptions.py

```diff
@@ -134,3 +134,14 @@ class DeviceDisconnectedError(AstarteError):
 
     def __init__(self, msg):
         self.msg = msg
+
+class DeviceGrpcDecodeAstarteDataError(AstarteError):
```

Comment:
I think this exception can be more generic. Like `DeviceGrpcDecodeError`. And then contain in the message that the error resulted from attempting to decode an Astarte Data. So that it can be re-used for other decoding errors.